src/stylesheets/components/_icon-list.scss

```diff
@@ -0,0 +1,152 @@
+// Variables
+
+// Size icons based on the theme-type-scale-[size] number and this factor
+$icon-list-icon-size-factor: 1.5;
+// Center the icon to the title text, works for all size variants
+$icon-list-icon-margin-top: -1.5%;
+// Fine tune the space between the icon and content with this factor
+$icon-list-icon-padding-left-factor: 0.4;
+// Fine tune the title top padding given a line height of 2
+$icon-list-title-padding-top: 0.2em;
+
+$icon-list-font-sizes: (
+  "xs" $theme-type-scale-xs,
+  "sm" $theme-type-scale-sm,
+  "md" $theme-type-scale-md,
+  "lg" $theme-type-scale-lg,
+  "xl" $theme-type-scale-xl,
+  "2xl" $theme-type-scale-2xl,
+  "3xl" $theme-type-scale-3xl
+);
+
+$type-scale: font-size($theme-icon-list-font-family, $theme-body-font-size);
+
+@include override-prose {
+  .usa-icon-list {
+    @include typeset($theme-icon-list-font-family);
+    @include unstyled-list;
+    @include u-measure(5);
+  }
+
+  .usa-icon-list__item {
+    display: flex;
+    font-size: size($theme-icon-list-font-family, $theme-icon-list-font-size);
+    position: relative;
+
+    & + .usa-icon-list__item {
+      padding-top: units(1.5);
+    }
+  }
+}
+
+// Allow setting icon color on the icon-list element
+.usa-icon-list {
+  @each $color, $grades in $all-project-colors {
+    @each $grade, $value in $grades {
+      @if $value {
+        $prefix: if($grade != "default", #{$color}-#{$grade}, #{$color});
+        &--#{$prefix} {
+          .usa-icon-list__icon {
+            color: color($prefix);
+          }
+        }
+      }
+    }
+  }
+}
+
+.usa-icon-list__icon {
+  .usa-icon {
+    // Set the height and width of the icon based on the size variant and factor
+    display: block;
+    height: $type-scale * $icon-list-icon-size-factor;
+    margin-top: $icon-list-icon-margin-top;
+    position: relative;
+    width: $type-scale * $icon-list-icon-size-factor;
+  }
+}
+
+// Generate responsive variants
+// Create a map for looping that includes a default or no-breakpoint version
+$this-null: (
+  "none": null,
+);
+$icon-list-breakpoints: map-merge($this-null, $system-breakpoints);
+@each $mq-key, $mq-value in $icon-list-breakpoints {
+  @each $token, $val in $icon-list-font-sizes {
+    // Set the prefix
+    // An empty string if "none"
+    $prefix: false;
+    @if $mq-key == "none" {
+      $prefix: "";
+    }
+    // Or the standard prefix if the breakpoint is output
+    @else if map-get($theme-utility-breakpoints, $mq-key) {
+      $prefix: "#{$mq-key}\\:";
+    }
+    @if $prefix {
+      $this-type-scale: font-size($theme-icon-list-font-family, $token);
+      @include at-media($mq-key) {
```

Comment:
Could we be setting this `@include at-media($mq-key)` higher up to prevent duplication in output?

<details>
  <summary>Example output</summary>
  
   ```css
@media all and (min-width: 30em){
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__icon .usa-icon{
    height:1.5rem;
    margin-top:-1.5%;
    width:1.5rem;
  }
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content{
    font-size:1rem;
    padding-left:0.4rem;
  }
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > p,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > h2,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > h3,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > h4,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > h5,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > h6,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > ul,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > ol,
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content > div{
    font-size:initial;
  }
  .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content .usa-icon-list__title{
    font-size:1rem;
  }
}
@media all and (min-width: 30em){
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__icon .usa-icon{
    height:1.59rem;
    margin-top:-1.5%;
    width:1.59rem;
  }
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content{
    font-size:1.06rem;
    padding-left:0.424rem;
  }
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > p,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > h2,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > h3,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > h4,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > h5,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > h6,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > ul,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > ol,
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content > div{
    font-size:initial;
  }
  .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content .usa-icon-list__title{
    font-size:1.06rem;
  }
}
@media all and (min-width: 30em){
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__icon .usa-icon{
    height:1.695rem;
    margin-top:-1.5%;
    width:1.695rem;
  }
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content{
    font-size:1.13rem;
    padding-left:0.452rem;
  }
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > p,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > h2,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > h3,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > h4,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > h5,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > h6,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > ul,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > ol,
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content > div{
    font-size:initial;
  }
  .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content .usa-icon-list__title{
    font-size:1.13rem;
  }
}
   ```

Reply:
OK, I think i've got this one done

Reply:
I still see duplicate media queries for different icon list sizes.

Something like this might fix it:
```scss
@each $mq-key, $mq-value in $icon-list-breakpoints {
  $prefix: false;

  @if $mq-key == "none" {
    $prefix: "";
  }
  // Or the standard prefix if the breakpoint is output
  @else if map-get($theme-utility-breakpoints, $mq-key) {
    $prefix: "#{$mq-key}\\:";
  }

  @include at-media($mq-key) {
    @each $token, $val in $theme-body-font-sizes {
      @if $prefix {
        $this-type-scale: font-size($theme-icon-list-font-family, $token);
        @debug($mq-key);
          .#{$prefix}usa-icon-list--size-#{$token} {
            .usa-icon-list__icon {
              .usa-icon {
                // Set the height and width of the icon based on the size variant and factor
                height: $this-type-scale * $icon-list-icon-size-factor;
                width: $this-type-scale * $icon-list-icon-size-factor;
              }
            }

            .usa-icon-list__content {
              @include u-measure(5);
              // Resize simple (un-marked up) content
              font-size: size($theme-icon-list-font-family, $token);
              // Calculate the space between the icon and content based on the size variant and factor
              padding-left: $this-type-scale * $icon-list-icon-padding-left-factor;

              .usa-icon-list__title {
                @include u-font($theme-icon-list-title-font-family, $token);
              }
            }
          }
      }
    }
  }
}
```

First I loop through the media queries and then I loop through the different sizes. Results in it being contained within a single media query.

<details>
  <summary>Sample output</summary>
  
  ```css
   @media all and (min-width: 30em){
    .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__icon .usa-icon{
      height:1.5rem;
      width:1.5rem;
    }
    .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content{
      max-width:72ex;
      font-size:1rem;
      padding-left:0.4rem;
    }
    .mobile-lg\:usa-icon-list--size-xs .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:0.91rem;
    }

    .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__icon .usa-icon{
      height:1.59rem;
      width:1.59rem;
    }
    .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content{
      max-width:72ex;
      font-size:1.06rem;
      padding-left:0.424rem;
    }
    .mobile-lg\:usa-icon-list--size-sm .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:0.98rem;
    }

    .mobile-lg\:usa-icon-list--size-md .usa-icon-list__icon .usa-icon{
      height:1.695rem;
      width:1.695rem;
    }
    .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content{
      max-width:72ex;
      font-size:1.13rem;
      padding-left:0.452rem;
    }
    .mobile-lg\:usa-icon-list--size-md .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:1.04rem;
    }

    .mobile-lg\:usa-icon-list--size-lg .usa-icon-list__icon .usa-icon{
      height:2.19rem;
      width:2.19rem;
    }
    .mobile-lg\:usa-icon-list--size-lg .usa-icon-list__content{
      max-width:72ex;
      font-size:1.46rem;
      padding-left:0.584rem;
    }
    .mobile-lg\:usa-icon-list--size-lg .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:1.34rem;
    }

    .mobile-lg\:usa-icon-list--size-xl .usa-icon-list__icon .usa-icon{
      height:3.195rem;
      width:3.195rem;
    }
    .mobile-lg\:usa-icon-list--size-xl .usa-icon-list__content{
      max-width:72ex;
      font-size:2.13rem;
      padding-left:0.852rem;
    }
    .mobile-lg\:usa-icon-list--size-xl .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:1.95rem;
    }

    .mobile-lg\:usa-icon-list--size-2xl .usa-icon-list__icon .usa-icon{
      height:3.99rem;
      width:3.99rem;
    }
    .mobile-lg\:usa-icon-list--size-2xl .usa-icon-list__content{
      max-width:72ex;
      font-size:2.66rem;
      padding-left:1.064rem;
    }
    .mobile-lg\:usa-icon-list--size-2xl .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:2.44rem;
    }

    .mobile-lg\:usa-icon-list--size-3xl .usa-icon-list__icon .usa-icon{
      height:4.785rem;
      width:4.785rem;
    }
    .mobile-lg\:usa-icon-list--size-3xl .usa-icon-list__content{
      max-width:72ex;
      font-size:3.19rem;
      padding-left:1.276rem;
    }
    .mobile-lg\:usa-icon-list--size-3xl .usa-icon-list__content .usa-icon-list__title{
      font-family:Merriweather Web, Georgia, Cambria, Times New Roman, Times, serif;
      font-size:2.93rem;
    }
  }
  ```
</details>

Reply:
ahh, I get you now! 🤦